lib/spack/spack/fetch_strategy.py

```diff
@@ -1102,6 +1143,12 @@ def _from_merged_attrs(fetcher, pkg, version):
 def for_package_version(pkg, version):
```

Comment:
Would something go wrong if `Package.fetcher` was updated to return `None` for a `BundlePackage`, vs. implementing a fetcher that always raises an error? If I understand correctly, the `do_fetch` function is not called at all for a `BundlePackage`.

If that creates many errors, I could imagine it's on account of cruft in the fetching logic and we could keep this (but I think in that case a TODO could be added here to refactor it out).

Reply:
FWIW. I overrode `BundlePackage`s fetcher before replacing with the fetch strategy.  I was concerned someone would question or complain about the approach.

Reply:
Stage creation (`_make_stage`) has issues wrt staging resources: `Construct a composite stage on top of the composite FetchStrategy`.  

Returning `None` for the fetcher causes problems processing resources (since it is not an iterable).

Returning `None` as the fetch strategy in `fs.for_package_version` yields an empty composite fetch strategy that results in the composite stage getting an index error during `source_path` creation.

Reply:
IMO updating the Stage to allow for an empty/null `FetchStrategy` would be better but I don't think that has to happen here: as long as `BundleFetchStrategy` is clear that it is a noop and there is a TODO to refactor.

Reply:
Changed relevant comments.